### ISSUE TITLE
Consider unify the sanitize inputs for every estimator 

### ISSUE THREAD

**Issue by CrystalXuR** [som-shahlab/survmetabench#93]

**Comment by CrystalXuR** [comment]
@erikcs Do you think these checks on the inputs at https://github.com/som-shahlab/survmetabench/blob/master/r-package/survlearners/R/surv_rl_grf.R#L86 are necessary? I adopted this from R-learner, but I don't have such checks in my other estimators, so I am thinking of either remove it for all or add it for all

**Comment by erikcs** [comment]
Yes input checks are necessary, there are some common ones like: 1) make sure all dims align in X,Y,W,D 2) they are numeric (e.g X can not be a string/factor for glmnet/grf...) 3) predict(object, newdata) newdata needs same dim as X.train that can be a separate function in utils

there are also estimator specific checks e.g. if all D=0 or all D=1 then your code will most likely break in places where you do a subset like `D==0` and continue using the result, or subset `W==1` if there are no treated (or too few treated if you make folds on that subset)

**Comment by erikcs** [comment]
we can do a meet first after the 1-6 in #22 is done

**Comment by CrystalXuR** [comment]
Sounds good, thanks
